pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CloseResourceRule.java

```diff
@@ -736,16 +736,21 @@ private boolean isAssignmentForVariable(ASTStatementExpression statement, ASTVar
         if (statement == null || variable == null) {
             return false;
         }
-        ASTName name = statement.getFirstDescendantOfType(ASTName.class);
-        if (name == null) {
-            return false;
-        }
-        NameDeclaration statementVariable = name.getNameDeclaration();
-        if (statementVariable == null) {
-            return false;
+
+        List<ASTAssignmentOperator> assignments = statement.findDescendantsOfType(ASTAssignmentOperator.class);
+        for (ASTAssignmentOperator assignment : assignments) {
+            // The sibling before the operator is the left hand side
+            JavaNode lhs = assignment.getParent().getChild(assignment.getIndexInParent() - 1);
+
+            ASTName name = lhs.getFirstDescendantOfType(ASTName.class);
```

Comment:
Not sure if this is important, but this `descendants` check could misinterpret `array[...] = ...`, because it will fetch the name of the array.

Might be worth figuring out a test case

Reply:
I've tested it a bit and I don't think, that's relevant: If the variable we search here would actually be an array, we wouldn't search it in the first place, because then it is not autocloseable.